activerecord/lib/active_record/connection_adapters/resolver.rb

```diff
@@ -53,6 +43,21 @@ def spec(config)
           raise AdapterNotFound, "database configuration specifies nonexistent #{db_config.adapter} adapter"
         end
 
+        db_config
+      end
+
+      # Returns an instance of ConnectionSpecification for a given adapter.
+      # Accepts a hash one layer deep that contains all connection information.
+      #
+      # == Example
+      #
+      #   config = { "production" => { "host" => "localhost", "database" => "foo", "adapter" => "sqlite3" } }
+      #   spec = Resolver.new(config).spec(:production)
+      #   spec.db_config.configuration_hash
+      #   # => { host: "localhost", database: "foo", adapter: "sqlite3" }
+      #
+      def spec(config)
+        db_config = lookup(config)
         ConnectionSpecification.new(db_config.configuration_hash.delete(:name) || "primary", db_config)
```

Comment:
Since the `resolve` call in `lookup` is no longer setting the `:name` key, I don't think it will even be there anymore.  If we were going to keep this method, then I think this line needs to be updated to just get the name from the `db_config` directly.  However, it looks like this method is considered private and this PR makes it unused, so we could just remove it entirely.